packages/ui/src/components/ListV2/HeaderCell.tsx

```diff
@@ -72,16 +72,27 @@ export const HeaderCell = ({
     order = 'descending'
   }
 
+  const handleOrder = onOrder
+    ? () => onOrder(order === 'ascending' ? 'desc' : 'asc')
+    : undefined
+
   return (
     <StyledHeaderCell
-      role={onOrder ? 'button columnheader' : 'columnheader'}
       aria-sort={order}
-      onClick={
-        onOrder
-          ? () => onOrder(order === 'ascending' ? 'desc' : 'asc')
+      className={className}
+      onClick={handleOrder}
+      onKeyDown={
+        handleOrder
+          ? event => {
+              if (event.key === ' ') {
```

Comment:
Maybe we should also map `enter` key?
As we use this as a button (https://www.w3.org/WAI/ARIA/apg/patterns/button/)